// sucrase/dist/HelperManager.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

const HELPERS = {
  require: `
    import {createRequire as CREATE_REQUIRE_NAME} from "module";
    const require = CREATE_REQUIRE_NAME(import.meta.url);
  `,
  interopRequireWildcard: `
    function interopRequireWildcard(obj) {
      if (obj && obj.__esModule) {
        return obj;
      } else {
        var newObj = {};
        if (obj != null) {
          for (var key in obj) {
            if (Object.prototype.hasOwnProperty.call(obj, key)) {
              newObj[key] = obj[key];
            }
          }
        }
        newObj.default = obj;
        return newObj;
      }
    }
  `,
  interopRequireDefault: `
    function interopRequireDefault(obj) {
      return obj && obj.__esModule ? obj : { default: obj };
    }
  `,
  createNamedExportFrom: `
    function createNamedExportFrom(obj, localName, importedName) {
      Object.defineProperty(exports, localName, {enumerable: true, configurable: true, get: () => obj[importedName]});
    }
  `,
  // Note that TypeScript and Babel do this differently; TypeScript does a simple existence
  // check in the exports object and does a plain assignment, whereas Babel uses
  // defineProperty and builds an object of explicitly-exported names so that star exports can
  // always take lower precedence. For now, we do the easier TypeScript thing.
  createStarExport: `
    function createStarExport(obj) {
      Object.keys(obj)
        .filter((key) => key !== "default" && key !== "__esModule")
        .forEach((key) => {
          if (exports.hasOwnProperty(key)) {
            return;
          }
          Object.defineProperty(exports, key, {enumerable: true, configurable: true, get: () => obj[key]});
        });
    }
  `,
  nullishCoalesce: `
    function nullishCoalesce(lhs, rhsFn) {
      if (lhs != null) {
        return lhs;
      } else {
        return rhsFn();
      }
    }
  `,
  asyncNullishCoalesce: `
    async function asyncNullishCoalesce(lhs, rhsFn) {
      if (lhs != null) {
        return lhs;
      } else {
        return await rhsFn();
      }
    }
  `,
  optionalChain: `
    function optionalChain(ops) {
      let lastAccessLHS = undefined;
      let value = ops[0];
      let i = 1;
      while (i < ops.length) {
        const op = ops[i];
        const fn = ops[i + 1];
        i += 2;
        if ((op === 'optionalAccess' || op === 'optionalCall') && value == null) {
          return undefined;
        }
        if (op === 'access' || op === 'optionalAccess') {
          lastAccessLHS = value;
          value = fn(value);
        } else if (op === 'call' || op === 'optionalCall') {
          value = fn((...args) => value.call(lastAccessLHS, ...args));
          lastAccessLHS = undefined;
        }
      }
      return value;
    }
  `,
  asyncOptionalChain: `
    async function asyncOptionalChain(ops) {
      let lastAccessLHS = undefined;
      let value = ops[0];
      let i = 1;
      while (i < ops.length) {
        const op = ops[i];
        const fn = ops[i + 1];
        i += 2;
        if ((op === 'optionalAccess' || op === 'optionalCall') && value == null) {
          return undefined;
        }
        if (op === 'access' || op === 'optionalAccess') {
          lastAccessLHS = value;
          value = await fn(value);
        } else if (op === 'call' || op === 'optionalCall') {
          value = await fn((...args) => value.call(lastAccessLHS, ...args));
          lastAccessLHS = undefined;
        }
      }
      return value;
    }
  `,
  optionalChainDelete: `
    function optionalChainDelete(ops) {
      const result = OPTIONAL_CHAIN_NAME(ops);
      return result == null ? true : result;
    }
  `,
  asyncOptionalChainDelete: `
    async function asyncOptionalChainDelete(ops) {
      const result = await ASYNC_OPTIONAL_CHAIN_NAME(ops);
      return result == null ? true : result;
    }
  `,
};

 class HelperManager {
  __init() {this.helperNames = {}}
  __init2() {this.createRequireName = null}
  constructor( nameManager) {;this.nameManager = nameManager;HelperManager.pros puede desbloquearlo por tiTus padres pueden desbloquearlo por tiAbre Chrome y accede a tu cuenta para que el programa pueda comprobar si puedes acceder a este sitio.Nuevo: Aquí se aplican las opciones de Family Link para ChromeLa configuración que eligió tu madre o padre ahora te protege en líneaEncriptar las contraseñas sincronizadas con tu Cuenta de GoogleContraseñasPestañas abiertasLista de lecturaNo se permite una frase de contraseña vacía.Opciones de encriptaciónEncriptar los datos sincronizados con tu propia frase de contraseña de sincronizaciónConfigurando…Frase de contraseñaDebes ingresar la misma frase de contraseña dos veces.Tus datos se encriptaron con la <a target="_blank" href="$1">frase de contraseña de sincronización</a> el $2. Ingresa la frase para iniciar la sincronización.Tus datos están encriptados con tu frase de contraseña para sincronización. Debes ingresarla para iniciar la sincronización.Más idiomas…¿La página no está en $1?Nunca traducir $1Nunca traducir este sitioSiempre traducir $1¿No está en $1? Informa este errorAcerca de Google TraductorTraducirEsta página se tradujo de$1a$2Esta página se tradujo al $1.Mostrar originalNo se puede traducir esta página.De ahora en más, las páginas en $1 se traducirán al $2.No se traducirán las páginas en $1.Este sitio no se traducirá.Idioma detectadoGrisAzulRojoAmarilloVerdeRosaPúrpuraCianNaranja&Deshacer&Rehacer&Deshacer Agregar&Rehacer Agregar&Deshacer Eliminar&Rehacer Eliminar&Deshacer Editar&Rehacer Editar&Deshacer Mover&Rehacer Mover&Deshacer Reorganizar&Rehacer Reorgan
